src/utils/installPackage.js

```diff
@@ -3,8 +3,9 @@ const config = require('./config');
 const path = require('path');
 const promisify = require('./promisify');
 const resolve = promisify(require('resolve'));
+const commandExists = require('command-exists').sync;
```

Comment:
This is used in an async method. Should use the promise version exposed by `command-exists` with `await`.

Reply:
Ya that’s exactly what I originally wanted to do that 😃 

But... because of the way the `command-exists` package is written, it’s doesn’t really work out. 😕 

Pretty much, `commandExists()` conveys the result of whether or not it found the command by resolving the returned promise for “package found”, and rejecting the promise for “package not found”. This is a problem if we’re using async/await because `await` handles promise rejections by throwing. So pretty much whenever yarn _isn’t_ found it’ll throw an error instead of nicely informing us.

I guess it’s possible to do it using a try/catch, but that’s really messy. The only other way to use the async version of `command-exists` is to just directly use the promise, but that’s really complicated since `promise.finally` hasn’t landed in Node yet and it would probably require nesting all the rest of the install code.

So I can still use the async version... just it won’t be as clean, and I don’t really think there’s that much of a performance loss from doing it synchronously, especially considering that the function it’s running in is gonna be executing asynchronously anyway. 

Your call.

Reply:
We're already inside of a promise chain (`return new Promise...`) so you could use the Promise API with a bit of refactoring. Alternatively, you could create a `hasYarnInstalled` method:

```js
hasYarnInstalled = async () => {
	try {
		return await commandExists('yarn');
	} catch (err) {
		return false;
	}
}

if (await hasYarnInstalled() && fs.exists(...)) {
   ...
}
```